mmv1/third_party/terraform/services/dns/data_source_dns_managed_zones.go

```diff
@@ -0,0 +1,242 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/api/dns/v1"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+
+	"github.com/hashicorp/terraform-provider-google/google/fwmodels"
+	"github.com/hashicorp/terraform-provider-google/google/fwresource"
+	"github.com/hashicorp/terraform-provider-google/google/fwtransport"
+)
+
+// Ensure the implementation satisfies the expected interfaces
+var (
+	_ datasource.DataSource              = &GoogleDnsManagedZonesDataSource{}
+	_ datasource.DataSourceWithConfigure = &GoogleDnsManagedZonesDataSource{}
+)
+
+func NewGoogleDnsManagedZonesDataSource() datasource.DataSource {
+	return &GoogleDnsManagedZonesDataSource{}
+}
+
+// GoogleDnsManagedZonesDataSource defines the data source implementation
+type GoogleDnsManagedZonesDataSource struct {
+	client  *dns.Service
+	project types.String
+}
+
+type GoogleDnsManagedZonesModel struct {
+	Id           types.String `tfsdk:"id"`
+	Project      types.String `tfsdk:"project"`
+	ManagedZones types.List   `tfsdk:"managed_zones"`
+}
+
+func (d *GoogleDnsManagedZonesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dns_managed_zones"
+}
+
+func (d *GoogleDnsManagedZonesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Provides access to all zones for a given project within Google Cloud DNS",
+
+		Attributes: map[string]schema.Attribute{
+
+			"project": schema.StringAttribute{
+				Description:         "The ID of the project for the Google Cloud.",
+				MarkdownDescription: "The ID of the project for the Google Cloud.",
+				Optional:            true,
+			},
+
+			// Id field is added to match plugin-framework migrated google_dns_managed_zone data source
+			// Whilst ID fields are required in the SDK, they're not needed in the plugin-framework.
+			"id": schema.StringAttribute{
+				Description:         "foobar",
+				MarkdownDescription: "foobar",
+				Computed:            true,
+			},
+		},
+
+		Blocks: map[string]schema.Block{
+			"managed_zones": schema.ListNestedBlock{
+				Description:         "The list of managed zones in the given project.",
+				MarkdownDescription: "The list of managed zones in the given project.",
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"name": schema.StringAttribute{
```

Comment:
This might be a framework thing, or behavior I'm just not aware of, but what is this used for? I noticed it's optional, and by name alone I was expecting to see some sort of filtering logic, which I couldn't find.

Reply:
Ah - the name and project fields for the elements in the `managed_zone` list should be Computed given that they're not affected by arguments passed to the data source. I think this was a copy-paste error from the singular data source. In future I'd like to work out how we can allow reusing and modifying schemas like how we allow SDK data sources to re-use resources' schemas (i.e swapping fields from Optional => Computed and changing Required true => false), but that's beyond the scope of this PR.

I'll swap these fields to be Computed only